src/python/antares_xpansion/input_checker.py

```diff
@@ -260,27 +316,106 @@ def check_candidates_file(driver):
+                    filename_path)
         if not has_a_profile:
             # remove candidate if it has no profile
             flushed_print("candidate %s will be removed!" %
                           ini_file[each_section]["name"])
             ini_file.remove_section(each_section)
             config_changed = True
 
-    if config_changed:
-        shutil.copyfile(driver.candidates_ini_filepath(),
-                        driver.candidates_ini_filepath() + ".bak")
-        with open(driver.candidates_ini_filepath(), 'w') as out_file:
-            ini_file.write(out_file)
-        flushed_print("%s file was overwritten! backup file %s created"
-                      % (driver.candidates_ini_filepath(), driver.candidates_ini_filepath() + ".bak"))
+    return config_changed
+
+
+def check_attributes_profile(ini_file, candidates_ini_filepath, capacity_dir_path):
+    # check attributes profile is 0, 1 or an existent filename
+    if check_attribute_profile_values(ini_file, capacity_dir_path):
+        copy_in_backup(ini_file, candidates_ini_filepath)
+
+
+def check_candidates_file(candidates_ini_filepath, capacity_dir_path):
+    """
+        checks that a candidate file related to an XpansionDriver has the correct format
+
+        :param driver: the XpansionDriver pointing to the candidates file
+
+        :return: Exits if the candidates files has the wrong format.
+    """
+    default_values = {'name': 'NA',
+                      'enable': 'true',
+                      'candidate-type': 'investment',
+                      'investment-type': 'generation',
+                      'link': 'NA',
+                      'annual-cost-per-mw': '0',
+                      'unit-size': '0',
+                      'max-units': '0',
+                      'max-investment': '0',
+                      'Relaxed': 'false',
+                      'link-profile': '1',
+                      'already-installed-capacity': '0',
+                      'already-installed-link-profile': '1'}
+    ini_file = configparser.ConfigParser(default_values)
+    ini_file.read(candidates_ini_filepath)
+
+    # check attributes types and values
+    check_candidate_attributes(ini_file)
+
+    # check that name is not empty and does not have space
+    # check that link is not empty
+    check_candidate_name_and_link(ini_file)
+
+    # check exclusion between max-investment and (max-units, unit-size) attributes
+    check_candidate_exclusive_attributes(ini_file)
+
+    # check attributes profile is 0, 1 or an existent filename
+    check_attributes_profile(
+        ini_file, candidates_ini_filepath, capacity_dir_path)
 
 
 ##########################################
 # Checks related to settings.ini
 ##########################################
+
+
+class NotHandledOption(Exception):
+    pass
+
+
+class NotHandledValue(Exception):
+    pass
+
+
+type_str = str
+type_int = int
+type_float = float
+
+# "option": (type, legal_value(s))
+options_types_and_legal_values = {
```

Comment:
There are also some deprecated options here :
- `method`
- `cut_type`
- `week_selection`
- `relaxed_optimality_gap`

For `master` the option `full_integer` is not valid any more.

Reply:
Allowed float values are in fact positive floats -> find a way to uniformize with the candidates positive float check